pkg/trace/agent/agent.go

```diff
@@ -138,6 +141,10 @@ type SpanModifier interface {
 	ModifySpan(*pb.TraceChunk, *pb.Span)
 }
 
+type TracerPayloadModifier interface {
+	ModifyTracerPayload(*pb.TracerPayload)
```

Comment:
I think we can simplify the name here, the type and params are sufficiently descriptive:
```suggestion
	Modify(*pb.TracerPayload)
```